client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -2462,4 +2462,25 @@ Undocumented.prototype.startMigration = function( sourceSiteId, targetSiteId ) {
 	} );
 };
 
+Undocumented.prototype.getAtomicSiteMediaViaProxy = function( siteIdOrSlug, mediaPath, fn ) {
+	return this.wpcom.req.get(
+		{
+			path: `/sites/${ siteIdOrSlug }/atomic-auth-proxy/file${ mediaPath }`,
+			apiNamespace: 'wpcom/v2',
+			responseType: 'blob',
+		},
+		fn
+	);
+};
+
+Undocumented.prototype.getAtomicSiteMediaViaProxyRetry = function( siteIdOrSlug, mediaPath, fn ) {
+	this.getAtomicSiteMediaViaProxy( siteIdOrSlug, mediaPath, function( err, data ) {
+		if ( err ) {
```

Comment:
I don't think this works as-is: when a request fails on the remote site, the response to the browser will still 200 & the `err` condition does not hit.

I'm working on a solution using the `data-layer` code that has built-in retrying, but it's not ready yet.